### DISCUSSION TITLE
base64 encoded jpeg

### DISCUSSION THREAD

**Discussion by xubury** [nothings/stb#1423] [Q&A]
Hi, I tried to load image with a base64 encoded string with my base64 decode function and `stbi_load_from_memory`, but failed with `unknown marker` or `expected marker`.  
However, when I write the decoded buffer to disk, it display normally with windows image displayer. 
So the base64 decode function should be fine.
I couldn't figure out what I do wrong.
The base64 encoded uri I used was read from a gltf file. 
Example here:
https://github.com/KhronosGroup/glTF-Sample-Models/blob/master/2.0/DamagedHelmet/glTF-Embedded/DamagedHelmet.gltf
Anyone has any ideas?
Heres my pseudo code:
```
      size_t size = base64Bytes(uri);
      void *data = malloc(size);
      decodeBase64(uri, data); // if i write this buffer to disk it display normally
      uint8_t *ptr = (uint8_t *)data;
      uint8_t *img = stbi_load_from_memory(ptr, size, &width, &height, &channels, 4);
      if (img == nullptr) log::error(stbi_failure_reason());
```

**Comment by nothings** [comment]
if you write the buffer to disk, and then load it with stbi_load, what happens?

**Reply by xubury** [reply]
It will return the same error, unknown marker or expected marker.

**Reply by nothings** [reply]
is the file different from the original image before you base64 coded it?

**Reply by xubury** [reply]
I don't have the exact original image before encoding, just the base64 string.
Although the gltf repo provides a non-base64 version, but I suspect they are the same since I can stbi_load them normally.

**Reply by nothings** [reply]
Doesn't that mean they're NOT the same?

**Reply by xubury** [reply]
Oh yeah, I meant they are not the same

**Comment by xubury** [comment]
I upload the minimum test code to repoduce.
It does 1. decode the string -> 2. stbi_load_from_memory -> 3. write deocded string to a jpeg file ->4. stbi_load the jpeg file
The step2 and step4 should print "unknown marker"
[stbtest.zip](https://github.com/nothings/stb/files/10292060/stbtest.zip)

**Reply by nothings** [reply]
I didn't try to run your program, but the test.jpeg in the zip file loads without error for me.

**Reply by xubury** [reply]
rellay? I try load it with these code, but it still failed with unknown marker. I am so confused right now.
```
  int32_t width, height, channels;
  uint8_t *img = stbi_load("test.jpeg", &width, &height, &channels, 4);
  if (img == nullptr) {
    printf("%s\n", stbi_failure_reason());
  }
  stbi_image_free(img);
```

**Reply by xubury** [reply]
I found a tool that can check the integrity of jpeg. 
It also gets the marker error: `test.jpeg 2048 x 2048 24bit JFIF  N  935629  Unsupported marker type 0xbf  [ERROR]`
